daemon/daemon.go

```diff
@@ -1168,6 +1168,13 @@ func checkKernel() error {
 	return nil
 }
 
+func (daemon *Daemon) adaptContainerSettings(hostConfig *runconfig.HostConfig) {
+	if hostConfig.Memory > 0 && hostConfig.MemorySwap == 0 {
+		// By default, MemorySwap is set to twice the size of Memory.
+		hostConfig.MemorySwap = hostConfig.Memory * 2
+	}
+}
+
```

Comment:
I'd say they are different things like @hqhq said. verifyContainerSettings() only sanitizes hostconfig when certain features are not available. This is more of policy settings.